app/Database/Requirement.hs

```diff
@@ -0,0 +1,59 @@
+{-|
+Description: Requirement
+
+Module containing data type that represents a "Requirement".
+
+We will use parsed data to create instances of this type.
+-}
+module Requirement
+( ProgramReq(..)
+, CourseRequirement(..)
+, CourseReq(..)
+, Req(..)
+, showProgramReq
+, showCourseRequirement
+, showCourseReq
+, showReq
+) where
+
+import Data.String
+
+
+data ProgramReq = PRGREQ String [Req]
+
+-- | Returns a well formatted String representing a program requirement for specified program.
+showProgramReq :: ProgramReq -> String
+showProgramReq (PRGREQ program reqs) = "The program requirements for " + program + ":\n"
+										+ map (showReq) reqs
+
+
+-- Assuming we have correctly parsed inputs, CourseReq is in order [corequisite, exclusion, prerequisite]
+data CourseRequirement = CRSREQ String [CourseReq]
+
+-- | Returns a well formatted String representing all course requirements for specified course.
+showCourseRequirement :: CourseRequirement -> String
```

Comment:
Read through [here](http://learnyouahaskell.com/making-our-own-types-and-typeclasses#typeclasses-102) to learn how to make `CourseRequirement`, and `Req` instances of the `Show` typeclass. (There's only one function you need to implement, `show`.)